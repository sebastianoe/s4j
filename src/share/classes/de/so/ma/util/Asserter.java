package de.so.ma.util;

import java.util.List;

/**
 * Utility class to provide some assertion shortcuts.
 * @author Sebastian Oergel
 *
 */
public class Asserter {
	/**
	 * Asserts that the list <code>params</code> contains exactly <code>n</code> items.
	 * @param params
	 * @param n
	 */
	public static void assertLenNParams(List<?> params, int n) {
		if (params.size() != n) {
			// tbd: better exception handling
			throw new RuntimeException("Length of params is not " + n);
		}
	}
	
	/**
	 * Asserts that the list <code>params</code> contains exactly 1 item.
	 * @param params
	 */
	public static void assertLen1List(List<?> params) {
		assertLenNParams(params, 1);
	}
}
