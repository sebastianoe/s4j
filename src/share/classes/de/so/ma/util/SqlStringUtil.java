package de.so.ma.util;

public class SqlStringUtil {
	public static String fixWhitespaces(String sqlString) {
		// remove newlines & tabs
		sqlString = sqlString.replaceAll("[\\n\\t]", "");
		
		// remove multiple spaces
		sqlString = sqlString.replaceAll("[ ]+", " ");
		
		return sqlString;
	}
	
	public static String stripName(String original) {
		return original.replaceAll("`", "");
	}
}
