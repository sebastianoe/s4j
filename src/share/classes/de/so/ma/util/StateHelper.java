package de.so.ma.util;

public class StateHelper {
	public static boolean usingJtreg() {
		return System.getProperty("test.src") != null;
	}
}
