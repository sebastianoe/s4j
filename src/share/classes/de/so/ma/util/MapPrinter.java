package de.so.ma.util;

import java.util.Map;
import java.util.Map.Entry;

import com.google.common.base.Strings;
import com.google.common.collect.Table;

import de.so.ma.data.ProjectionItem;
import de.so.ma.validation.matching.flatobject.FlatObjectAttribute;

public class MapPrinter<K, V> {
	public void printMap(Map<K, V> map, Table<ProjectionItem, FlatObjectAttribute, Integer> votings) {
		if (map == null) {
			System.out.println("No map to print");
			return;
		}
		
		int maxKeyLength = getMaxLength(map.keySet());
		int maxValueLength = getMaxLength(map.values());

		for (Entry<K, V> entry : map.entrySet()) {
			System.out.print(Strings.padStart(entry.getKey().toString(), maxKeyLength, ' '));
			System.out.print(" --> ");
			System.out.print(Strings.padEnd(entry.getValue().toString(), maxValueLength, ' '));
			System.out.println(" (" + votings.get(entry.getKey(), entry.getValue()) + ")");
			System.out.println();
		}
	}

	private int getMaxLength(Iterable<?> iterable) {
		int maxLength = 0;

		for (Object o : iterable) {
			int currentLength = o.toString().length();
			if (currentLength > maxLength) {
				maxLength = currentLength;
			}
		}

		return maxLength;
	}

}
