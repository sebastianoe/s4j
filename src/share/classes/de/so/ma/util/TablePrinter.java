package de.so.ma.util;

import java.util.List;

import com.google.common.base.Strings;
import com.google.common.collect.Lists;
import com.google.common.collect.Table;

public class TablePrinter<R,C,V> {
	public void printTable(Table<R, C, V> table) {
		List<C> columns = Lists.newArrayList(table.columnKeySet());
		int[] columnLengths = getColumnLengths(table);
		
		// print column headers
		System.out.print(Strings.padStart("", columnLengths[0], ' '));
		for (int i = 1; i <= columns.size(); i++) {
			System.out.print(Strings.padStart(columns.get(i - 1).toString(), columnLengths[i], ' '));
			System.out.print("|");
		}
		
		// newline
		System.out.println();
		
		// print rows
		for (R row : table.rowKeySet()) {
			System.out.print(Strings.padStart(row.toString(), columnLengths[0], ' '));
			for (int i = 1; i <= columns.size(); i++) {
				System.out.print(
						Strings.padStart(
								table.get(row, columns.get(i - 1)).toString(), columnLengths[i], ' '));
				System.out.print("|");
			}
			
			// newline
			System.out.println();
		}
	}

	private int[] getColumnLengths(Table<R, C, V> table) {
		int[] lengths = new int[table.columnKeySet().size() + 1];
		List<C> columns = Lists.newArrayList(table.columnKeySet());
		
		for (R row : table.rowKeySet()) {
			int rowLength = row.toString().length();
			if (rowLength > lengths[0]) {
				lengths[0] = rowLength;
			}
			
			for (int i = 1; i <= columns.size(); i++) {
				int columnLength = columns.get(i - 1).toString().length();
				if (columnLength > lengths[i]) {
					lengths[i] = columnLength;
				}
			}
		}
		
		return lengths;
	}
	
	
	
	
}
