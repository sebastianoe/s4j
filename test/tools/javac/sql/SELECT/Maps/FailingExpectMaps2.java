/*
 * @test  /nodynamiccopyright/
 * @summary Failing Maps 2
 * @compile/fail/ref=FailingExpectMaps2.out FailingExpectMaps2.java -sqlrepo /Users/sebastian/ma/jdk7u-langtools/test/tools/javac/sql/repo.json
 */

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.HashMap;
import com.google.common.collect.Lists;
import com.google.common.collect.Iterables;

public class FailingExpectMaps2 {
    public static void main(String[] args) {
    	Map<String, String> sqlMap = SQL[SELECT city, name, id FROM storage];
    }    
}