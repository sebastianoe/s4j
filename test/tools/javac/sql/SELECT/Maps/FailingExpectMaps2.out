error: Query must contain exactly 2 projection to be assignable to a map.
1 error